### PR TITLE
Modify the rewrite plugin to write multiple EDNS0 options (#936)

### PR DESCRIPTION
**Pull request by huynhmb359**
### 1. What does this pull request do?
Modify the rewrite plugin to write multiple EDNS0 options if multiple is specified.  For below example, 
3 edns0 options would be inserted.  

~~~
.:53 {
  proxy . 10.82.3.15:53 
  rewrite continue edns0 subnet set 24 56
  rewrite continue edns0 local set 0xfff3 {client_ip}
  rewrite continue edns0 local set 0xfff4 {protocol} 
}
~~~

### 2. Which issues (if any) are related?
Issue #936 

### 3. Which documentation changes (if any) need to be made?
README

### CODE REVIEW

File: plugin/rewrite/type.go [hunk 1]
```diff
@@ -35,3 +35,8 @@ func (rule *typeRule) Rewrite(w dns.ResponseWriter, r *dns.Msg) Result {
 	}
 	return RewriteIgnored
 }
+
+// Mode returns the processing mode
+func (rule *typeRule) Mode() string {
+	return "stop"
```
Comment (miekg) [inline plugin/rewrite/type.go]: nit; shouldn't this be `Stop` the constant?

File: plugin/rewrite/name.go [hunk 2]
```diff
@@ -22,3 +22,8 @@ func (rule *nameRule) Rewrite(w dns.ResponseWriter, r *dns.Msg) Result {
 	}
 	return RewriteIgnored
 }
+
+// Mode returns the processing mode
+func (rule *nameRule) Mode() string {
+	return "stop"
```
Comment (miekg) [inline plugin/rewrite/name.go]: constant Stop here as well?

**Review by miekg — APPROVED** [review]

**Comment by johnbelamaric** [comment]
Please update with the constant as Miek mentioned, then I will merge.

File: plugin/rewrite/type.go (hunk 1 above)
Comment (huynhmb359) [reply plugin/rewrite/type.go]: oops,  let me fix it.

File: plugin/rewrite/README.md [hunk 3]
```diff
@@ -26,8 +26,11 @@ needs to be a full match of the name, e.g., `rewrite name miek.nl example.org`.
 
 When the FIELD is `edns0` an EDNS0 option can be appended to the request as described below.
 
-If you specify multiple rules and an incoming query matches on multiple (simple) rules, only
-the first rewrite is applied.
+If you specify multiple rules and an incoming query matches on multiple (simple) rules, the rewrite
+will behave as following
+* `continue` will continue apply the next rule in the rule list. 
+* `stop` will consider the current rule is the last rule and will not continue.  Default behaviour
+for not specificy this rule processing mode is `stop`
```
Comment (johnbelamaric) [inline plugin/rewrite/README.md]: s/specificy/specifying/

File: plugin/rewrite/README.md [hunk 4]
```diff
@@ -26,8 +26,11 @@ needs to be a full match of the name, e.g., `rewrite name miek.nl example.org`.
 
 When the FIELD is `edns0` an EDNS0 option can be appended to the request as described below.
 
-If you specify multiple rules and an incoming query matches on multiple (simple) rules, only
-the first rewrite is applied.
+If you specify multiple rules and an incoming query matches on multiple (simple) rules, the rewrite
```
Comment (johnbelamaric) [inline plugin/rewrite/README.md]: s/(simple)//

We don't have any other type yet.

**Review by johnbelamaric — CHANGES_REQUESTED** [review]

**Review by johnbelamaric — APPROVED** [review]

**Pull request merged**